### PR TITLE
Ignore SV submitted txs in activity record computation

### LINKED ISSUE #5409: Do not count SV submissions in app activity recording
### Problem

A featured app creating a contract that is then archived   by a transaction submitted by an SV node will create featured app activity twice for the featured app: once for the tx creating the contract, and once for the SV-submitted transaction archiving it.

The latter is problematic as there is no burn for an SV-submitted transaction.

### Proposal

Do not count app activity for transactions submitted by SV-nodes. 

Concretely:

1. Store `DsoRules` in `RewardReferenceStore` and bump its key to ensure reingestion starting.
2. Do not create app activity for submissions from participants whose `participantId` is in `DsoRules.svs`. 
3. Bump the "code version" of app meta table for app activity computation to avoid mismatches between SVs.
4. Call out the change and the reset of app activity records in the release notes.

### Discussion

FYI: @rautenrieth-da @dfordivam @adetokunbo

### PR DESCRIPTION
**Pull request by dfordivam**
Fixes #5409 

#### Cluster Testing
- [ ] If a cluster test is required, comment `/cluster_test` on this PR to request it, and ping someone with access to the DA-internal system to approve it.
- [ ] If a hard-migration test is required (from the latest release), comment `/hdm_test` on this PR to request it, and ping someone with access to the DA-internal system to approve it.
- [ ] If a logical synchronizer upgrade test is required (from canton-3.5), comment `/lsu_test` on this PR to request it, and ping someone with access to the DA-internal system to approve it.

#### PR Guidelines
- [ ] Include any change that might be observable by our partners or affect their deployment in the [release notes](https://github.com/canton-network/splice/blob/main/docs/src/release_notes.rst).
- [ ] Specify fixed issues with `Fixes #n`, and mention issues worked on using `#n`
- [ ] Include a screenshot for frontend-related PRs - see [README](https://github.com/canton-network/splice/blob/main/TESTING.md#running-and-debugging-integration-tests) or use your favorite screenshot tool

#### Merge Guidelines
- [ ] Make the git commit message look sensible when squash-merging on GitHub (most likely: just copy your PR description).

### CODE REVIEW

File: apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/DbScanRewardsReferenceStore.scala [hunk 1]
```diff
@@ -110,6 +112,27 @@ class DbScanRewardsReferenceStore(
     lookupFeaturedAppRightsAsOf(asOf)
       .map(_.map(_.contract.payload.provider).toSet)
 
+  override def lookupSvParticipantIdsAsOf(
+      asOf: CantonTimestamp
+  )(implicit tc: TraceContext): Future[Set[String]] =
+    tcsStore.listAllContractsAsOf(DsoRules.COMPANION, asOf).map {
```
Comment (meiersi-da) [inline apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/DbScanRewardsReferenceStore.scala]: Consider using a query that lists only the first of these contracts, and thus uses a `LIMIT 1`, which allows for better query plans, even if the statistics are off. I'd be OK giving up the check on "multiple" once we've understood the problem of there being too many that you are currently observing.

Or alternatively, do a list all but with a limit of 2 contracts, and thus keep both the right query plan and the invariant check.

File: apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/TrafficBasedRewardsTimeBasedIntegrationTest.scala [hunk 2]
```diff
@@ -185,6 +204,20 @@ class TrafficBasedRewardsTimeBasedIntegrationTest
...
+      assertTrafficSummary(event, "aliceCreateId")
+      assertAppActivity(event, "aliceCreateId", Set(aliceParty), expectedRound = 7)
+    }
+
+    clue("SV-submitted expire TransferInstruction creates no app activity for alice") {
+      val event = fetchEvent(svExpireId, "svExpireId")
+      event.verdict shouldBe defined
+      assertTrafficSummary(event, "svExpireId")
+      assertNoAppActivity(event, "svExpireId")
+    }
```
Comment (meiersi-da) [inline apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/TrafficBasedRewardsTimeBasedIntegrationTest.scala:219]: nice. thanks for going the whole way and testing this.

**Review by meiersi-da — APPROVED** [review]
Very good work! Thank you.

Feel free to merge once you 
- [ ] added release notes
- [ ] have a green CI

**Comment by dfordivam** [comment]
Please have another look. This PR now includes fixes for two bugs I discovered for some edge cases.
  - Two DsoRules in store was caused by missing `delete` in
    `clearDataForCurrentMigrationId`. In the Simtime tests the time resets to 0,
    and this could lead to DsoRules added from two different runs being present in the archive table.
  - The "No DsoRules found" issue was actually caused by a side-effect of adding ingestion of
    DsoRules in DbScanRewardsReferenceStore. Now its possible that the DsoRules
    get archived before the OpenMiningRound does (due to SV onboarding in test
    initialization).
    As the oldest row in archive table acts as the "ingestionStart" for the store.
    The DsoRules is archived multiple times during SVs onboarding while the rounds are still active.
    This can result in oldest active OpenMiningRound (round 0) having openAt before the
    actual `ingestionStart` time of the SVs who join later.
    The fix is simple, just ignore events if round's `openAt` is before `ingestionStart`.

File: apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/DbMultiDomainAcsStore.scala [hunk 3]
```diff
@@ -1197,10 +1197,18 @@ final class DbMultiDomainAcsStore[TXE](
...
           "clearDataForCurrentMigrationId.deleteReassignments",
         )
+        nArchive <- acsArchiveConfigOpt match {
+          case Some(AcsArchiveConfig(archiveTableName, _)) =>
+            storage.update(
+              sqlu"delete from #$archiveTableName where store_id = $acsStoreId and migration_id = $domainMigrationId",
+              "clearDataForCurrentMigrationId.deleteArchive",
+            )
+          case None => FutureUnlessShutdown.pure(0)
+        }
```
Comment (meiersi-da) [inline apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/DbMultiDomainAcsStore.scala:1207]: Given this bug, we might also have SV nodes with a borked SvRewardsStore. wdyt about bumping the store key, so it gets reingested. We actually anyways have to do that to trigger ingestion of the `DsoRules`.

File: docs/src/release_notes_upcoming.rst [hunk 4]
```diff
@@ -38,3 +38,17 @@
...
+        as the SVs don't burn traffic and transactions submitted by them should not generate
+        traffic-based app rewards as specified in CIP-0104.
+
+        The data provided by the experimental ``app_activity_records`` field of ``GET /v0/events/{update-id}``
+        and ``POST /v0/events`` endpoints prior to this release may have attributed app-activity to
+        SV submitted transactions, and such app activity records should be considered incorrect.
+        App activity records data provided for other transactions is valid.
+
+        The ``app_activity_record_store`` table has been truncated to remove the records computed with the earlier logic,
+        and both the endpoints will not provide app ``app_activity_records`` for events prior to this release.
```
Comment (meiersi-da) [inline docs/src/release_notes_upcoming.rst:54]: Good release notes! Thank you!

**Review by meiersi-da — COMMENTED** [review]
Thanks for the fixes! Good work!
